pkg/network/bridge/plugin.go

```diff
@@ -114,27 +122,40 @@ func (nbdp *NetworkBridgeDevicePlugin) Allocate(ctx context.Context, r *pluginap
 
 	for _, req := range r.ContainerRequests {
 		var devices []*pluginapi.DeviceSpec
+		var vnics []vnic
 		for _, nic := range req.DevicesIDs {
 			dev := new(pluginapi.DeviceSpec)
 			assignmentPath := getAssignmentPath(nbdp.bridge, nic)
 			dev.HostPath = fakeDevicePath
 			dev.ContainerPath = assignmentPath
 			dev.Permissions = "r"
 			devices = append(devices, dev)
+			vnics = append(vnics, vnic{nic, protocolEthernet})
 
 			nbdp.assignmentCh <- &Assignment{
 				nic,
 				assignmentPath,
 				time.Now(),
 			}
 		}
+		envs := map[string]string{
+			envVariableName: marshal(nbdp.bridge, vnics),
+		}
 		response.ContainerResponses = append(response.ContainerResponses, &pluginapi.ContainerAllocateResponse{
-			Devices: devices})
+			Devices: devices, Envs: envs,
+		})
+
 	}
 
 	return &response, nil
 }
 
+func marshal(interfaceName string, vnics []vnic) string {
+	vnicsPerInterface := map[string][]vnic{interfaceName: vnics}
+	marshalled, _ := json.Marshal(vnicsPerInterface)
```

Comment:
you should add error handling to the json.Marshal call. so probably better to move that logic up into the calling function

Reply:
done